### DISCUSSION TITLE
No new versions - how we carry forward?

### DISCUSSION THREAD

**Discussion by BananaSupreme** [rajanadar/VaultSharp#255] [General]
Hi,

It seems the library is no longer maintained, pull requests are being made by some good individuals but no new versions are released as @rajanadar  seems to be missing since Nov...

Also having one person responsible for versioning and being high level admin might seem risky if he doesn't come back.

Ideas on what can we do?

**Comment by hoerup** [comment]
Last time @rajanadar went silent a colleague and I did make a fork. After he re-appeared we deleted the fork and I got permissions to merge PR's - and i think i can create releases as well. I do NOT have perms on appveyor or nuget so im unsure whether i can upload a new release

**Reply by BananaSupreme** [reply]
that seems to be the major issue, we can't really update the version, and who knows when he'll be back.
I want to contribute, but it feels kinda pointless if it won't really release, maybe we should create a secondry that has a more distributed ownership... or some sort of fail safe

**Comment by hoerup** [comment]
I have updated change log and tagged a new release. 

Unfortunately it looks like @rajanadar  have not set up release/publish workflows in AppVeyor - so i have no means to publish to nuget.org

What we *could* try to do, is setup GH actions and make it publish to GH package repo ??

**Reply by BananaSupreme** [reply]
Can we make a v2 package? Maybe it will take some time to catch on, but
then we can also make some refactorings and changes, mark nullability...
We can use gh packages but it wont update nuget, it will just be available
for anyone who is familiar with this package

EDIT - I wanted to clarify I don't mean making a new nuget right away, I mean planning for the possibility so maybe making a road map that allows breaking changes so we can reexamine some of the design choices that were made, and maybe allow for modernization of the code to improve its speed and usability. but that's a bit of a long term thing. For now starting GH actions and uploading to the git repo can be great, and when we finish if @rajanadar  is not back yet, we can further look at options

**Comment by rajanadar** [comment]
@hoerup  @BananaSupreme  Let me publish a couple of versions, one today with transit features and another within the week for remaining open issues. And then, take you up on this auto publishing via app-veyor so that the sole dependency on me is removed. Appreciate your suggestions.

**Comment by EKovalenok** [comment]
Is there any news on when the 1.13 version will be released?

**Comment by rajanadar** [comment]
hi @EKovalenok , what feature are you looking for?

Currently, I am waiting for Vault 1.13 release so that VaultSharp 1.13 can be released.
I tend to keep the VaultSharp versions equal or less than the released Vault versions, not more.

If you're looking for a specific feature, let me know. I'll do a 1.12.x release, just with that.
Otherwise, you can use the latest VaultSharp release, 1.12.2 (published just a few days ago)

**Reply by EKovalenok** [reply]
I would like to get a version of the package with GH-293 and GH-291 features. Not an urgent question, but will be very relevant to me in a week or two

[GH-291](https://github.com/rajanadar/VaultSharp/issues/291 ) Add logger endpoints to System backend.
[GH-293](https://github.com/rajanadar/VaultSharp/issues/291 ) secret/kv2: Add new fields cas_required and custom_metadata in FullSecretMetadata class.

**Reply by rajanadar** [reply]
@EKovalenok , 293 sounds good.

I think 291 is not yet supported in any of the released Vault APIs. They themselves plan to support it in 1.13 only. My tests were failing on 1.12.2 of Vault. I'll double check again. So wasn't sure, how you'll make use of that feature if Vault itself hasn't released that feature.

**Comment by rajanadar** [comment]
@EKovalenok , i have published 1.12.2.1 of https://www.nuget.org/packages/VaultSharp 
It has what you need
